### PR TITLE
fix: Showing the additional structuring in cap confirmation

### PR DESCRIPTION
**Pull request by NitishAroraKPMG**
## Description

Showing the additional structuring in the cap confirmation

## Testing instructions

Create a capped ticket by the distance, taps, and price and check for additional structuring int the cap confirmation page

### CODE REVIEW

File: repos/fdbt-site/src/pages/capConfirmation.tsx [hunk 1]
```diff
@@ -233,6 +247,13 @@ export const getServerSideProps = async (ctx: NextPageContextWithSession): Promi
                 : 'Rolling days'
             : '';
 
+    const additionalPricing =
+        typeOfCapAttribute.typeOfCap === 'byDistance'
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/capConfirmation.tsx:251]: do we need to check the type of cap? I think we can just see if the additionalPricingAttribute is there and use the presence of that?

File: repos/fdbt-site/src/pages/capConfirmation.tsx [hunk 2]
```diff
@@ -233,6 +247,13 @@ export const getServerSideProps = async (ctx: NextPageContextWithSession): Promi
...
+            ? additionalPricingAttribute && 'pricingStructureStart' in additionalPricingAttribute
+                ? `Pricing structure starts after ${additionalPricingAttribute.pricingStructureStart} min with percentage discount ${additionalPricingAttribute.structureDiscount} %`
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/capConfirmation.tsx:253]:
```suggestion
                ? `Pricing structure starts after ${additionalPricingAttribute.pricingStructureStart} min with percentage discount ${additionalPricingAttribute.structureDiscount}%`
```

File: repos/fdbt-site/src/pages/capConfirmation.tsx (hunk 1 above)
Comment (alainajaffer) [reply repos/fdbt-site/src/pages/capConfirmation.tsx:251]: I don't think this can be done as the attribute is optional

Comment (RobCattonKPMG) [reply repos/fdbt-site/src/pages/capConfirmation.tsx:251]: ^ incorrect

**Review by alainajaffer — DISMISSED** [review]

File: repos/fdbt-site/src/pages/capConfirmation.tsx [hunk 3]
```diff
@@ -233,6 +247,13 @@ export const getServerSideProps = async (ctx: NextPageContextWithSession): Promi
                 : 'Rolling days'
             : '';
 
+    const additionalPricing =
+        typeOfCapAttribute.typeOfCap === 'byDistance'
+            ? additionalPricingAttribute && 'pricingStructureStart' in additionalPricingAttribute
+                ? `Pricing structure starts after ${additionalPricingAttribute.pricingStructureStart} min with percentage discount ${additionalPricingAttribute.structureDiscount}%`
```
Comment (alainajaffer) [inline repos/fdbt-site/src/pages/capConfirmation.tsx:253]: change back

**Review by alainajaffer — APPROVED** [review]

File: repos/fdbt-site/tests/pages/capConfirmation.test.tsx [hunk 4]
```diff
@@ -75,6 +77,7 @@ describe('pages', () => {
                     capDistancePricingContents={['Min price: £2, Max price: £4']}
                     distanceBands={['0 km - End of journey, Price per km: £4']}
                     csrfToken=""
+                    additionalPricing="Pricing structure starts after 2 min with percentage discount 2 %"
```
Comment (alainajaffer) [inline repos/fdbt-site/tests/pages/capConfirmation.test.tsx:80]: may need to be changed because of the space

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**